lib/data/repositories/trips_repo_impl.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:dartz/dartz.dart';
+import 'package:injectable/injectable.dart';
+import 'package:trip_tally/data/dto/trips/add_trip_dto.dart';
+import 'package:trip_tally/domain/data_source/trips_data_source.dart';
+import 'package:trip_tally/domain/entities/add_trip_entity/add_trip_entity.dart';
+import 'package:trip_tally/domain/repositories/trips_repo.dart';
+import 'package:trip_tally/domain/utils/failure.dart';
+import 'package:trip_tally/domain/utils/success.dart';
+import 'package:trip_tally/presentation/utils/enums/errors.dart';
+
+@Injectable(as: TripsRepo)
+class TripsRepoImpl implements TripsRepo {
+  TripsRepoImpl(this._dataSource);
+
+  final TripsDataSource _dataSource;
+
+  @override
+  Future<Either<Failure, Success>> newTrip(AddTripEntity entity) async {
```

Comment:
```suggestion
  Future<Either<Failure, Success>> createTrip(AddTripEntity entity) async {
```
keep one naming convention on layers